extensions/ql-vscode/src/extension.ts

```diff
@@ -879,25 +882,34 @@ async function activateWithInstalledDistribution(
     );
 
   void extLogger.log("Initializing QLTest interface.");
-  const testExplorerExtension = extensions.getExtension<TestHub>(
-    testExplorerExtensionId,
-  );
-  let testUiCommands: Partial<TestUICommands> = {};
-  if (testExplorerExtension) {
-    const testHub = testExplorerExtension.exports;
-    const testAdapterFactory = new QLTestAdapterFactory(
-      testHub,
-      cliServer,
-      dbm,
-    );
-    ctx.subscriptions.push(testAdapterFactory);
 
-    const testUIService = new TestUIService(app, testHub);
-    ctx.subscriptions.push(testUIService);
+  const testRunner = new TestRunner(dbm, cliServer);
+  ctx.subscriptions.push(testRunner);
+
+  let testManager: TestManagerBase | undefined = undefined;
+  if (isCanary()) {
```

Comment:
Hmm. There's no way to "unregister" a `TestController` AFAICT, but maybe just deleting all of its `TestItem`s would be equivalent. I'll have to see if there's a way to unregister the legacy test adapter, but it might not be worth the effort.

Reply:
OK, there is a way to unregister the legacy test stuff. However, handling of the commands we expose is trickier. We share the same command IDs between the two implementations, but they route to slightly different implementations depending on which test UI we're using.

Given how easy the transition to the new UI was, and how annoying our dependency on that third-party extension is, I suspect we'll move this out of canary pretty soon anyway.

Reply:
If it's going to take significant time and still be imperfect, I don't think it's worth taking on for a transitional feature.